File: src/main/java/org/bakkes/game/controller/input/CompositeKeyListener.java
package org.bakkes.game.controller.input;

import java.util.Collection;

import com.google.inject.Inject;

public class CompositeKeyListener implements IKeyListener{

	private @Inject Collection<IKeyListener> keyListeners;
	@Override
	public void KeyDown(final Key key) {
		for(final IKeyListener k : keyListeners){
			k.KeyDown(key);
		}
	}

	@Override
	public void KeyUp(final Key key) {
		for(final IKeyListener k : keyListeners){
			k.KeyUp(key);
		}
	}
}

File: src/main/scala/org/bakkes/game/controller/state/event/input/CompositeKeyListener.scala
package org.bakkes.game.controller.state.event.input

import java.util.Collection

import com.google.inject.Inject

import scala.collection.JavaConversions._

class CompositeKeyListener @Inject() (listeners : Collection[IKeyListener]) extends IKeyListener{

	override def KeyDown(key:Key) = {
		listeners.foreach{_.KeyDown(key)}
	}

	override def KeyUp(key:Key) = {
		listeners.foreach{_.KeyUp(key)}
	}

	def clear() = {
		listeners.clear();
	}
	def add(listner:IKeyListener) : Boolean = {
		return listeners.add(listner);
	}
}
